scanpipe/pipes/d2d.py

```diff
@@ -774,94 +777,163 @@ def _map_javascript_resource(
+            if mapped_resources_about:
+                mapped_resources_about.append(to_resource)
+            else:
+                self.mapped_resources_by_aboutpath[about_path] = [to_resource]
+            to_resource.update(status=flag.ABOUT_MAPPED)
+
+    def create_about_packages_relations(self, project):
+        for about_path, mapped_resources in self.mapped_resources_by_aboutpath.items():
+            about_file_resource = self.about_resources_by_path[about_path]
+            package_data = self.about_pkgdata_by_path[about_file_resource.path]
+
+            if not mapped_resources:
+                error_message_details = {
+                    "path": about_file_resource.path,
+                    "package_data": package_data,
+                }
+                project.add_warning(
+                    description=(
+                        "Resource paths listed at about_resource is not found"
+                        " in the to/ codebase"
+                    ),
+                    model="map_about_files",
+                    details=error_message_details,
+                )
+                continue
 
-    codebase_resources.update(status=flag.ABOUT_MAPPED)
-    about_file_resource.update(status=flag.ABOUT_MAPPED)
+            # Create the Package using .ABOUT data and assign related codebase_resources
+            pipes.update_or_create_package(project, package_data, mapped_resources)
+
+            # Map the .ABOUT file resource to all related resources in the ``to/`` side.
+            for mapped_resource in mapped_resources:
+                pipes.make_relation(
+                    from_resource=about_file_resource,
+                    to_resource=mapped_resource,
+                    map_type="about_file",
+                )
+
+            about_file_resource.update(status=flag.ABOUT_MAPPED)
+
+        for about_file_resource in self.about_resources_by_path.values():
+            about_file_companions = (
+                about_file_resource.siblings()
+                .filter(name__startswith=about_file_resource.name_without_extension)
+                .filter(extension__in=[".LICENSE", ".NOTICE"])
+            )
+            about_file_companions.update(status=flag.ABOUT_MAPPED)
 
 
 def map_about_files(project, logger=None):
     """Map ``from/`` .ABOUT files to their related ``to/`` resources."""
     project_resources = project.codebaseresources
-    from_files = project_resources.files().from_codebase()
-    from_about_files = from_files.filter(extension=".ABOUT")
-    to_resources = project_resources.to_codebase()
+    from_about_files = (
+        project_resources.files().from_codebase().filter(extension=".ABOUT")
+    )
+    if not from_about_files.exists():
+        return
+
+    indexes = AboutFileIndexes.create_indexes(
+        project=project, from_about_files=from_about_files
+    )
+
+    to_resources = project_resources.to_codebase().no_status()
```

Comment:
I am not sure we should exclude things based on  `no status` as this makes this super specific to the step ordering.... at least we should add a comment here.